python-web-applications/main.py

```diff
@@ -0,0 +1,35 @@
+from flask import Flask
+from flask import request
+
+app = Flask(__name__)
+
+
+def convert_temp(cel_temp):
+    """Converts Celsius temperature to Fahrenheit temperature."""
+    try:
+        far_temp = float(cel_temp) * 9 / 5 + 32
+        far_temp = round(far_temp, 3)  # Round to three decimal places
+        return str(far_temp)
+    except ValueError:  # User entered non-numeric temperature
+        return "invalid input"
+
+
+@app.route("/")
+def index():
+    cel_temp = request.args.get("cel_temp")
+    if cel_temp:
+        far_temp = convert_temp(cel_temp)
+    else:
+        far_temp = ""
+    return (
+        """<form action="">
+                Celsius: <input type="text" name="cel_temp">
+                <input type="submit" value="Convert">
+            </form>"""
+        + "Fahrenheit: "
+        + far_temp
+    )
+
+
+if __name__ == "__main__":
+    app.run(host="127.0.0.1", port=8080, debug=True)
```

Comment:
It's great that you use the standard black formatting for the supplementing code, unlike the code that we show in the article. However, I'd use more descriptive names and drop a few comments:

```suggestion
from flask import Flask
from flask import request

app = Flask(__name__)


def fahrenheit_from(celsius):
    """Convert Celsius to Fahrenheit degrees."""
    try:
        fahrenheit = float(celsius) * 9 / 5 + 32
        fahrenheit = round(fahrenheit, 3)  # Round to three decimal places
        return str(fahrenheit)
    except ValueError:
        return "invalid input"


@app.route("/")
def index():
    celsius = request.args.get("celsius", "")
    if celsius:
        fahrenheit = fahrenheit_from(celsius)
    else:
        fahrenheit = ""
    return (
        """<form action="" method="get">
                Celsius temperature: <input type="text" name="celsius">
                <input type="submit" value="Convert to Fahrenheit">
            </form>"""
        + "Fahrenheit: "
        + fahrenheit
    )


if __name__ == "__main__":
    app.run(host="127.0.0.1", port=8080, debug=True)

```

Reply:
sure, I agree with that. I left most of the code the way it was, but why not also update the code example, not just the text : )

I'll propagate the changes into the code snippets in the article as well.